src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -80,6 +92,46 @@ def spark_version_compatibility(spark_version: str) -> str:
     return "supported"
 
 
+class AzureServicePrincipalCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "azure_service_principals")
+        self._ws = ws
+
+    def _crawl(self) -> list[AzureServicePrincipalInfo]:
+        all_service_principals = list(self._ws.service_principals.list())
```

Comment:
This issue is about service principals used to access data, not ones that tun clusters

Reply:
added logic in def _assess_pipelines(self, all_pipelines):. Can you please review. If ok, then I will add the same in a module for other place - such as cluster ,job cluster, etc.

Reply:
the service principals that are used to access data for users are not added as users.